src/wallet/wallet.h

```diff
@@ -477,21 +477,21 @@ class CWalletTx : public CMerkleTx
 
 class CInputCoin {
 public:
-    CInputCoin()
-    {
-    }
     CInputCoin(const CWalletTx* walletTx, unsigned int i)
     {
-        if (walletTx != nullptr && i < walletTx->tx->vout.size())
+        if (walletTx == nullptr)
+            throw std::invalid_argument("walletTx should not be null");
+        if (i < walletTx->tx->vout.size())

```

Comment:
Ah indeed, C# habits... Just replaced by `if(!walletTx)`.

I would like to do `const CWalletTx&` in a separate PR, as it need more code change, and I need this PR for another one depending on it.